main.go

```diff
@@ -1,95 +1,22 @@
 package main
 
 import (
-	"encoding/json"
 	"fmt"
+	"log"
 	"net/http"
-)
-
-// Collection is a struct that does something
-type Collection struct {
-	ID        int    `json:"id"`
-	Name      string `json:"name"`
-	Publisher string `json:"published"`
-}
-
-type Volume struct {
-	ID           int    `json:"id"`
-	CollectionID int    `json:"collection_id"`
-	Name         string `json:"name"`
-	Edition      string `json:"edition"`
-	Pages        int    `json:"pages"`
-}
-
-var collectionList = []Collection{
-	{1, "Turma da Mônica", "X"},
-	{2, "Asterix", "Y"},
-}
-
-var volumesList = []Volume{
-	{1, 1, "Almanacão", "1", 39},
-	{2, 1, "Especial de natal", "especial", 20},
-	{3, 2, "Asterix em São Paulo", "3", 120},
-	{4, 2, "Asterix en français", "francesa", 2},
-}
-
-func collectionHandler(w http.ResponseWriter, r *http.Request) {
-	jsonList, err := json.Marshal(collectionList)
-	if err != nil {
-		panic(err)
-	}
-
-	w.Write([]byte(jsonList))
-}
 
-// FindVolumes is a function that...
-func (collection Collection) FindVolumes() []Volume {
-	result := []Volume{}
-	for _, volume := range volumesList {
-		if volume.CollectionID == collection.ID {
-			result = append(result, volume)
-		}
-	}
-	return result
-}
-
-func volumesHandler(w http.ResponseWriter, r *http.Request) {
-	// fmt.Println("asdasd")
-	// urlPathSegments := strings.Split(r.URL.Path, "/volumes")
-	// fmt.Println(string(urlPathSegments[0][1:]))
-	// collectionID, err := strconv.Atoi(string(urlPathSegments[0][1:]))
-	// if err != nil {
-	// 	w.WriteHeader(http.StatusNotFound)
-	// 	return
-	// }
-
-	collectionID := 2
+	"hq-collections.com/collection"
+	"hq-collections.com/volume"
+)
 
-	emptyCollection := Collection{}
-	currentCollection := emptyCollection
-	for _, item := range collectionList {
-		if item.ID == collectionID {
-			currentCollection = item
-		}
-	}
+const basePath = "/api"
 
-	if currentCollection == emptyCollection {
-		w.WriteHeader(http.StatusNotFound)
-		return
-	}
-
-	jsonList, err := json.Marshal(currentCollection.FindVolumes())
+func main() {
+	collection.SetupRoutes(basePath)
+	volume.SetupRoutes(basePath)
+	fmt.Println("Webserver is up and running, waiting for connections...")
+	err := http.ListenAndServe(":5000", nil)
```

Comment:
@arturcp shouldn't we relate the http + collection + volume routes altogether? Like a single instance of http? I could not grasp how this works.

Reply:
I feel it should be in one place as well. I guess they do like this because each package has its own handler, but we could try to move the routes to another place and import the handlers somehow.

Reply:
I guess we don't have to use the same `http` because what it is doing in the SetupRoute is telling "if the request follows this pattern, use this handler". It is a generic thing, it seems